src/sage/matroids/oriented_matroids/oriented_matroid.py

```diff
@@ -0,0 +1,737 @@
+        """
+        return self._groundset
+
+    def elements(self):
+        """
+        Return all elements.
+
+        The elements of an oriented matroid are the "defining" elements of
+        the oriented matroid. For example, covectors are the elements of
+        an oriented matroid defined using covectors.
+        """
+        return self._elements
+
+    def circuits(self):
+        """
+        Return all circuits.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: M = OrientedMatroid([[1], [-1]], key='circuit')
+            sage: M.circuits()
+            [+: 0
+             -:
+             0: ,
+             +:
+             -: 0
+             0: ]
+        """
+        if hasattr(self, "_circuits"):
+            return self._circuits
+        raise NotImplementedError("circuits not implemented")
+
+    def cocircuits(self):
+        """
+        Return all cocircuits.
+        """
+        if hasattr(self, "_cocircuits"):
+            return self._cocircuits
+        raise NotImplementedError("cocircuits not implemented")
+
+    def vectors(self):
+        """
+        Return all vectors.
+        """
+        if hasattr(self, "_vectors"):
+            return self._vectors
+        raise NotImplementedError("vectors not implemented")
+        pass
+
+    def covectors(self):
+        """
+        Return all covectors.
+        """
+        if hasattr(self, "_covectors"):
+            return self._covectors
+        raise NotImplementedError("covectors not implemented")
+
+    def convert_to(self, new_type=None):
+        '''
+        Return an oriented matroid of type specified.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: M = OrientedMatroid([[1], [-1], [0]], key='vector')
+            sage: M.convert_to('circuit')
+            Circuit oriented matroid of rank 0
+            sage: M.convert_to()
+            Traceback (most recent call last):
+            ...
+            TypeError: must be given a type to convert to
+        '''
+        from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+        if new_type is None:
+            raise TypeError("must be given a type to convert to")
+        elif new_type in OrientedMatroid.keys:
+            if hasattr(self, new_type + 's'):
+                els = getattr(self, new_type + 's')()
+            else:
+                raise NotImplementedError("no %ss() method found in oriented matroid" % (new_type,))
+            return OrientedMatroid(els,
+                                   key=new_type,
+                                   groundset=self.groundset())
+        else:
+            raise NotImplementedError("type %s not implemented" % (new_type,))
+
+    def dual(self):
+        """
+        Return the dual oriented matroid.
+
+        EXAMPLES::
```

Comment:
```suggestion
        """
        Return the dual oriented matroid.

        EXAMPLES::
```
You should add an example for the case when there is no dual. If no such case exists, then this should be an `@abstract_method` (perhaps with `optional=True`).

Reply:
Turns out that dual is not actually implemented yet. I had missed that an (incorrect) doctest was written by one of my mentees. I'll go ahead and fix this and see about adding code for making a dual.